src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java

```diff
@@ -66,21 +66,22 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
 
         boolean ignoreContrastFindings = Boolean.parseBoolean(build.getBuildVariableResolver().resolve("ignoreContrastFindings"));
 
-        String applicationId = getApplicationId(contrastSDK, profile.getOrgUuid(), build.getParent().getDisplayName());
-        if (applicationId.equals("")) {
-            VulnerabilityTrendHelper.logMessage(listener, "Application with name '" + build.getParent().getDisplayName() + "' not found.");
-            if (profile.isFailOnWrongApplicationName()) {
-                throw new AbortException("Application with name '" + build.getParent().getDisplayName() + "' not found.");
-            }
-        }
-
         // iterate over conditions; fail on first
         for (ThresholdCondition condition : conditions) {
+
+            String applicationId = getApplicationId(contrastSDK, profile.getOrgUuid(), condition.getApplicationName());
+            if (applicationId.equals("")) {
```

Comment:
Is it possible to add some tests around this? These are tricky conditions.

Reply:
In all of the unsuccessful cases, the returned app id will be empty string. In case the app id is wrong, or if exceptions occur, the returned applicationId will be empty string.

Reply:
Great. Add some tests around it to make sure.

Reply:
Done, added "VulnerabilityTrendRecorderTest".

Reply:
Good start. We need tests for conditions on lines 86, 90, 95, 104 as a start

Reply:
@gmoore Tests added except for line 95. No actual connection is made to the teamServer, so the Unauthorized exception will not be thrown.